client/cli/src/main.rs

```diff
@@ -428,99 +477,51 @@ struct PayRequest {
 async fn main() {
     TracingSetup::default().init().expect("tracing initializes");
 
-    if let Ok(cli) = CliNoWorkdir::try_parse() {
-        // Only commands that don't need the workdir can be used here
-        //TODO: remove allow when there are more commands
-        #[allow(irrefutable_let_patterns)]
-        if let CommandNoWorkdir::VersionHash = cli.command {
-            println!(
-                "{}",
-                CliOutput::VersionHash {
-                    hash: env!("CODE_VERSION").to_string()
-                }
-            );
-        };
-    } else {
-        let module_gens = ClientModuleGenRegistry::from(vec![
-            DynClientModuleGen::from(WalletClientGen),
-            DynClientModuleGen::from(MintClientGen),
-            DynClientModuleGen::from(LightningClientGen),
-        ]);
-
-        let cli = Cli::parse();
-        if let Command::JoinFederation { connect } = cli.command {
+    let cli = Opts::parse();
+
+    match handle_command(cli).await {
+        Ok(output) => {
+            // ignore if there's anyone reading the stuff we're writting out
+            let _ = writeln!(std::io::stdout(), "{output}");
+        }
+        Err(err) => {
+            let _ = writeln!(std::io::stderr(), "{err}");
+            exit(1);
+        }
+    }
+}
+
+async fn handle_command(cli: Opts) -> CliResult {
+    let mut task_group = TaskGroup::new();
+    let mut rng = rand::rngs::OsRng;
+
+    match cli.command.clone() {
+        Command::JoinFederation { connect } => {
```

Comment:
Nothing special about this sub-command, so it's now handled like everything else.